wicket-orientdb/src/main/java/ru/ydn/wicket/wicketorientdb/utils/query/StringQueryManager.java

```diff
@@ -19,33 +19,30 @@
 	private static final Pattern PROJECTION_PATTERN = Pattern.compile("select\\b(.+?)\\bfrom\\b", Pattern.CASE_INSENSITIVE);
 	private static final Pattern EXPAND_PATTERN = Pattern.compile("expand\\((.+)\\)", Pattern.CASE_INSENSITIVE);
     private static final Pattern ORDER_CHECK_PATTERN = Pattern.compile("order\\s+by", Pattern.CASE_INSENSITIVE);
-    
+	private static final Pattern EMBEDDED_PATTERN = Pattern.compile("\\(select\\b(.+?)\\bfrom\\b(.+)\\)", Pattern.CASE_INSENSITIVE);
+
     private String projection;
     private boolean containExpand;
     private boolean hasOrderBy;
     private String sql;
     private String countSql;
     private final Map<String, IFilterCriteriaManager> managers;
-    
+
     public StringQueryManager(String sql) {
     	this.sql = sql;
     	Matcher matcher = PROJECTION_PATTERN.matcher(sql);
-        if(matcher.find())
-        {
+        if(matcher.find()) {
         	projection = matcher.group(1).trim();
         	Matcher expandMatcher = EXPAND_PATTERN.matcher(projection);
         	containExpand = expandMatcher.find();
-        	if(containExpand)
-        	{
-        		countSql = matcher.replaceFirst("select sum("+expandMatcher.group(1)+".size()) as count from");
```

Comment:
Not sure that this is right. Turning to select count(*) will require loading of full array and will work slower rather than just getting size of the array.